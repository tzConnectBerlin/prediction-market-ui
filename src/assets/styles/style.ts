import { css, SerializedStyles } from '@emotion/react';
import { Theme } from '@material-ui/core';

export const GlobalStyle = (theme: Theme): SerializedStyles => css`
  body {
    margin: 0;
    font-family: 'Roboto', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
  .MuiAlert-standardError {
    color: ${theme.palette.error.main};
    background-color: ${theme.palette.error.dark};
  }
  .MuiAlert-standardSuccess {
    color: ${theme.palette.success.main};
    background-color: ${theme.palette.success.dark};
  }
  .MuiAlert-standardWarning {
    color: ${theme.palette.warning.main};
    background-color: ${theme.palette.warning.dark};
  }
  .MuiFormControl-root {
    margin-top: ${theme.spacing(1)};
    margin-bottom: ${theme.spacing(1)};
    width: 100%;
    button[disabled],
    .MuiButton-root.Mui-disabled {
      background-color: ${theme.palette.primary.main};
      color: ${theme.palette.buttonText.disabled};
    }
    .MuiFormHelperText-root {
      color: ${theme.palette.text.secondary};
      &:not(.extra-help-message) {
        color: ${theme.palette.warning.main};
      }
    }
    .MuiInput-root.MuiInputBase-formControl {
      &:before,
      &.Mui-disabled::before,
      &:hover:not(.Mui-disabled)::before,
      &:after {
        border-bottom: none;
      }
      background-color: ${theme.palette.grey[400]};
      border: solid 1px ${theme.palette.grey[600]};
      border-radius: ${theme.spacing(1 / 4)};
      padding-left: ${theme.spacing(1 / 2)};
      padding-right: ${theme.spacing(1 / 2)};
      &:not(.Mui-disabled) {
        &:hover {
          background-color: ${theme.palette.grey[500]};
        }
        &:focus {
          border-color: ${theme.palette.primary.main};
        }
      }
      &.Mui-error {
        border-color: ${theme.palette.warning.main};
      }
      .MuiSelect-select {
        padding-left: ${theme.spacing(1 / 2)};
      }
      .MuiSelect-icon {
        color: ${theme.palette.text.primary};
      }
    }
  }
  .MuiFormLabel-root.MuiInputLabel-root,
  .MuiInputLabel-formControl.MuiInputLabel-root {
    font-weight: bold;
    color: ${theme.palette.primary.main};
    position: relative;
    transform: none;
    transition: none;
    font-size: 0.8rem;
    &.Mui-disabled {
      color: ${theme.palette.primary.main};
      opacity: 0.38;
    }
    & .label-asterisk {
      vertical-align: sub;
      line-height: 0;
    }
  }
  .MuiSlider {
    &-root {
      color: ${theme.palette.primary.main};
      &.Mui-disabled {
        color: ${theme.palette.primary.main};
        opacity: 0.38;
      }
    }
    &-thumb {
      background-color: ${theme.palette.primary.main};
      &.Mui-disabled {
        background-color: ${theme.palette.primary.main};
      }
    }
  }
`;
