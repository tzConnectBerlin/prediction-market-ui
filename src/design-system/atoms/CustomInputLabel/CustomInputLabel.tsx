import styled from '@emotion/styled';
import { InputLabel, InputLabelProps, TextFieldProps, Theme } from '@material-ui/core';

interface StyledInputLabelProps extends InputLabelProps {
  /**
   * Material-UI Theme
   */
  theme?: Theme;
  /**
   * Font weight to provide (default: 600)
   */
  fontWeight?: number | string;
  /**
   * Margin top (default: -12%)
   */
  marginTop?: number | string;
  /**
   * Custom class to add to asterisk (default: label-asterisk)
   */
  asteriskClass?: string;
}
export interface CustomInputLabelProps extends StyledInputLabelProps {
  /**
   * Show asterisk or not
   */
  required?: boolean;
  /**
   * Text to show
   */
  label: TextFieldProps['label'];
}

const StyledInputLabel = styled(InputLabel)<StyledInputLabelProps>`
  margin-top: ${({ marginTop }) => marginTop ?? '-12%'};
  font-weight: ${({ fontWeight }) => fontWeight ?? 600};
  & .label-asterisk {
    vertical-align: sub;
  }
  color: ${({ theme }) => (theme && theme.palette ? theme.palette.primary.main : 'inherit')};
`;

export const CustomInputLabel: React.FC<CustomInputLabelProps> = ({
  label,
  required,
  asteriskClass = 'label-asterisk',
  ...rest
}) => {
  return (
    <StyledInputLabel required={required} classes={{ asterisk: asteriskClass }} {...rest}>
      {label}
    </StyledInputLabel>
  );
};
