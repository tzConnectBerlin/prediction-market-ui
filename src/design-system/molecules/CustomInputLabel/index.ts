export type { CustomInputLabelProps } from './CustomInputLabel';
export type { CustomInputChipProps } from './CustomInputLabel';
export { CustomInputLabel, CustomInputLabel as default } from './CustomInputLabel';
