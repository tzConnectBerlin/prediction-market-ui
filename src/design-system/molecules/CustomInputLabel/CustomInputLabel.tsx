import styled from '@emotion/styled';
import { InputLabel, InputLabelProps, TextFieldProps, Theme, Grid } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import { CustomTooltip, CustomTooltipProps } from '../../atoms/CustomTooltip/CustomTooltip';

interface StyledInputLabelProps extends InputLabelProps {
  /**
   * Material-UI Theme
   */
  theme?: Theme;
  /**
   * Font weight to provide (default: 600)
   */
  fontWeight?: number | string;
  /**
   * Margin top (default: -12%)
   */
  marginTop?: number | string;
  /**
   * Custom class to add to asterisk (default: label-asterisk)
   */
  asteriskClass?: string;
}
export interface CustomInputLabelProps extends StyledInputLabelProps {
  /**
   * Show asterisk or not
   */
  required?: boolean;
  /**
   * Text to show
   */
  label: TextFieldProps['label'];
  tooltipProps?: CustomTooltipProps;
}

const StyledInputLabel = styled(InputLabel)<StyledInputLabelProps>`
  margin-top: ${({ marginTop }) => marginTop ?? '-12%'};
  font-weight: ${({ fontWeight }) => fontWeight ?? 600};
  color: ${({ theme }) => (theme && theme.palette ? theme.palette.primary.main : 'inherit')};
  &.Mui-disabled {
    color: ${({ theme }) => (theme && theme.palette ? theme.palette.primary.main : 'inherit')};
    opacity: 0.38;
  }
  & .label-asterisk {
    vertical-align: sub;
  }
`;

export const CustomInputLabel: React.FC<CustomInputLabelProps> = ({
  label,
  required,
  asteriskClass = 'label-asterisk',
  disabled,
  tooltipProps,
  ...rest
}) => {
  const theme = useTheme();
  return (
    <Grid container>
      <Grid item xs={11}>
        <StyledInputLabel
          required={required}
          classes={{ asterisk: asteriskClass }}
          theme={theme}
          disabled={disabled}
          {...rest}
        >
          {label}
        </StyledInputLabel>
      </Grid>
      {tooltipProps && (
        <Grid xs={1} item marginTop={rest.marginTop ?? '-12%'}>
          <CustomTooltip {...tooltipProps} />
        </Grid>
      )}
    </Grid>
  );
};
