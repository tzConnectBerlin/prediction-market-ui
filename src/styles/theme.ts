import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(255, 255, 255, 1)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    light: 'rgba(1, 102, 255, 0.6)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.06)',
    '300': 'rgba(29, 34, 39, 0.04)',
    '200': 'rgba(29, 34, 39, 0.03)',
    '100': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.65)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.65)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
  typography: {
    fontWeightRegular: 400,
    fontWeightBold: 700,
    h1: {
      fontWeight: 700,
      fontSize: '1.875rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h2: {
      fontWeight: 700,
      fontSize: '1.125rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h3: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h4: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h5: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h6: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    subtitle1: {
      fontWeight: 700,
      fontSize: '0.75rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
  zIndex: {
    tooltip: 9999,
  },
});
