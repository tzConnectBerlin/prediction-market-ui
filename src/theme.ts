import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
    light: 'rgba(212, 241, 204, 0.12)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.04)',
    '300': 'rgba(29, 34, 39, 0.03)',
    '200': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.54)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.54)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
  // action: {
  //   disabledBackground: "#0166FF",
  //   disabled: "rgba(255,255,255,0.3)",
  // },
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
});

export const darkTheme = createTheme({
  palette: {
    ...lightThemePalette,
    primary: {
      dark: 'rgba(214, 231, 255, 1)',
      main: 'rgba(204, 224, 255, 1)',
      light: 'rrgba(204, 224, 255, .12)',
      contrastText: 'rgba(29, 34, 39, 0.87)',
    },
    secondary: {
      dark: 'rgba(204, 224, 255, 0.14)',
      main: 'rgba(204, 224, 255, 0.12)',
      light: 'rgba(204, 224, 255, 0.14)',
      contrastText: 'rgba(204, 224, 255, 1)',
    },
    tertiary: {
      dark: 'rgba(212, 241, 204, 1)',
      main: 'rgba(212, 241, 204, 1)',
      light: 'rgba(212, 241, 204, 0.12)',
    },
    grey: {
      '900': 'rgba(255, 255, 255, 1)',
      '800': 'rgba(255, 255, 255, 0.87)',
      '700': 'rgba(255, 255, 255, 0.38)',
      '600': 'rgba(255, 255, 255, 0.2)',
      '500': 'rgba(255, 255, 255, 0.11)',
      '400': 'rgba(255, 255, 255, 0.09)',
      '300': 'rgba(255, 255, 255, 0.03)',
      '200': 'rgba(255, 255, 255, 0.02)',
    },
    text: {
      primary: 'rgba(255, 255, 255, 0.87)',
      secondary: 'rgba(255, 255, 255, 0.54)',
      disabled: 'rgba(255, 255, 255, 0.38)',
    },
    buttonText: {
      primary: 'rgba(29, 34, 39, 0.87)',
      secondary: 'rgba(29, 34, 39, 0.54)',
      disabled: 'rgba(29, 34, 39, 0.38)',
    },
    background: {
      default: 'rgba(29, 34, 39, 1)',
      paper: 'rgba(45, 49, 54, 1)',
    },
    mode: 'dark',
  },
  zIndex: {
    tooltip: 9999,
  },
});
