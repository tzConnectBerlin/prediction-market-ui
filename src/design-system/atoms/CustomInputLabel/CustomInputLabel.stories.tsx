import { Story, Meta } from '@storybook/react/types-6-0';
import { ThemeProvider } from '@material-ui/core/styles';
import { theme } from '../../../theme';
import { CustomInputLabel, CustomInputLabelProps } from './CustomInputLabel';

export default {
  title: 'Atom/CustomInputLabel',
  component: CustomInputLabel,
} as Meta;

const Template: Story<CustomInputLabelProps> = (args) => (
  <ThemeProvider theme={theme}>
    <CustomInputLabel {...args} />
  </ThemeProvider>
);

export const Default = Template.bind({});
Default.args = {
  label: 'Enter a question',
  theme,
  marginTop: '2%',
};

export const Required = Template.bind({});
Required.args = {
  label: 'Enter a question',
  theme,
  marginTop: '2%',
  required: true,
};

export const WithoutTheme = Template.bind({});
WithoutTheme.args = {
  label: 'Enter a question',
  marginTop: '2%',
  required: true,
};
