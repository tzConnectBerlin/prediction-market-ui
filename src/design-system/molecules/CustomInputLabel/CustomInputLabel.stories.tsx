import * as React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { CustomInputLabel, CustomInputLabelProps } from './CustomInputLabel';

export default {
  title: 'Molecule/CustomInputLabel',
  component: CustomInputLabel,
} as Meta;

const Template: Story<CustomInputLabelProps> = (args) => <CustomInputLabel {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'Enter a question',
};

export const Required = Template.bind({});
Required.args = {
  label: 'Enter a question',
  required: true,
};
