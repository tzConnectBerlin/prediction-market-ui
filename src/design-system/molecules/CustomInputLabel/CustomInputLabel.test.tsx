import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import { CustomInputLabelProps, CustomInputLabel } from './CustomInputLabel';
import { lightTheme as theme } from '../../../styles/theme';

const onlyRequiredProps: CustomInputLabelProps = {
  label: 'Create a question',
};

const defaultProps: CustomInputLabelProps = {
  ...onlyRequiredProps,
};

const WrappedComponent: React.FC = (args: any) => (
  <ThemeProvider theme={theme}>
    <CustomInputLabel {...args} />
  </ThemeProvider>
);

describe('Snapshot testing CustomInputLabel Component', () => {
  it('renders correctly with default props', () => {
    const inputLabel = renderer.create(<WrappedComponent {...defaultProps} />).toJSON();
    expect(inputLabel).toMatchSnapshot();
  });

  it('renders correctly with only required props', () => {
    const inputLabel = renderer.create(<WrappedComponent {...onlyRequiredProps} />).toJSON();
    expect(inputLabel).toMatchSnapshot();
  });

  it('renders correctly with tooltip', () => {
    const inputLabel = renderer
      .create(<WrappedComponent tooltip tooltipText="tooltip" {...defaultProps} />)
      .toJSON();
    expect(inputLabel).toMatchSnapshot();
  });
});

describe('Element testing CustomInputLabel Component', () => {
  it('render correctly CustomInputLabel with default props', async () => {
    const { getByText } = render(<WrappedComponent {...defaultProps} />);
    expect(getByText(/Create a question/i)).toBeInTheDocument();
  });

  it('render correctly CustomInputLabel with only required props', async () => {
    const { getByText } = render(<WrappedComponent {...defaultProps} />);
    expect(getByText(/Create a question/i)).toBeInTheDocument();
  });

  it('render without theme provider', async () => {
    const { getByText } = render(<CustomInputLabel {...defaultProps} />);
    expect(getByText(/Create a question/i)).toBeInTheDocument();
  });
});
