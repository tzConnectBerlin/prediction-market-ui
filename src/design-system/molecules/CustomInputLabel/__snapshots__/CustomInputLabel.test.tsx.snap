// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing CustomInputLabel Component renders correctly with default props 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-basis: 91.666667%;
  -ms-flex-preferred-size: 91.666667%;
  flex-basis: 91.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 91.666667%;
}

.emotion-2 {
  color: rgba(29, 34, 39, 0.54);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-2.Mui-focused {
  color: rgba(1, 102, 255, 1);
}

.emotion-2.Mui-disabled {
  color: rgba(29, 34, 39, 0.38);
}

.emotion-2.Mui-error {
  color: rgba(255, 57, 57, 1);
}

<div
  className="MuiGrid-root MuiGrid-container emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 emotion-1"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary emotion-2"
    >
      Create a question
    </label>
  </div>
</div>
`;

exports[`Snapshot testing CustomInputLabel Component renders correctly with only required props 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-basis: 91.666667%;
  -ms-flex-preferred-size: 91.666667%;
  flex-basis: 91.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 91.666667%;
}

.emotion-2 {
  color: rgba(29, 34, 39, 0.54);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-2.Mui-focused {
  color: rgba(1, 102, 255, 1);
}

.emotion-2.Mui-disabled {
  color: rgba(29, 34, 39, 0.38);
}

.emotion-2.Mui-error {
  color: rgba(255, 57, 57, 1);
}

<div
  className="MuiGrid-root MuiGrid-container emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 emotion-1"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary emotion-2"
    >
      Create a question
    </label>
  </div>
</div>
`;

exports[`Snapshot testing CustomInputLabel Component renders correctly with tooltip 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-basis: 91.666667%;
  -ms-flex-preferred-size: 91.666667%;
  flex-basis: 91.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 91.666667%;
}

.emotion-2 {
  color: rgba(29, 34, 39, 0.54);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-2.Mui-focused {
  color: rgba(1, 102, 255, 1);
}

.emotion-2.Mui-disabled {
  color: rgba(29, 34, 39, 0.38);
}

.emotion-2.Mui-error {
  color: rgba(255, 57, 57, 1);
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-basis: 8.333333%;
  -ms-flex-preferred-size: 8.333333%;
  flex-basis: 8.333333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 8.333333%;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 12px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 0;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4:hover {
  background-color: rgba(0, 0, 0, 0);
}

.emotion-5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-6 {
  height: 1rem;
  width: 1rem;
}

.emotion-6 circle {
  fill: rgba(1, 102, 255, 1);
}

.emotion-6 circle:first-of-type {
  opacity: 0.16;
}

.emotion-6 rect {
  fill: rgba(1, 102, 255, 1);
}

<div
  className="MuiGrid-root MuiGrid-container emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 emotion-1"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary emotion-2"
    >
      Create a question
    </label>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 emotion-3"
  >
    <button
      aria-label="tooltip"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  emotion-4"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label emotion-5"
      >
        <svg
          className="emotion-6"
          color="rgba(1, 102, 255, 1)"
          enableBackground="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "rgba(1, 102, 255, 1)",
            }
          }
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="24"
            cy="24"
            fill="#2196F3"
            r="21"
          />
          <rect
            fill="#fff"
            height="11"
            width="4"
            x="22"
            y="22"
          />
          <circle
            cx="24"
            cy="16.5"
            fill="#fff"
            r="2.5"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
